fastlane/spec/actions_specs/verify_build_spec.rb

```diff
@@ -0,0 +1,96 @@
+describe Fastlane do
+  describe Fastlane::FastFile do
+    describe "verify_build" do
+      let(:no_such_file) { "no-such-file.ipa" }
+      let(:not_an_ipa) { File.expand_path("./fastlane_core/spec/fixtures/ipas/not-an-ipa.ipa") }
+      let(:correctly_signed_ipa) { File.expand_path("./fastlane_core/spec/fixtures/ipas/very-capable-app.ipa") }
+      let(:incorrectly_signed_ipa) { File.expand_path("./fastlane_core/spec/fixtures/ipas/ContainsWatchApp.ipa") }
+
+      before(:each) do
+        Fastlane::Actions.lane_context[Fastlane::Actions::SharedValues::IPA_OUTPUT_PATH] = nil
+        allow(FastlaneCore::UI).to receive(:success).with("Driving the lane 'test' 🚀")
+      end
+
+      if FastlaneCore::Helper.mac?
+        it "uses the ipa output path from lane context" do
+          Fastlane::Actions.lane_context[Fastlane::Actions::SharedValues::IPA_OUTPUT_PATH] = correctly_signed_ipa
+
+          expect(FastlaneCore::UI).to receive(:success).with("Build is verified, have a 🍪.")
+          Fastlane::FastFile.new.parse("lane :test do
+                verify_build
+            end").runner.execute(:test)
+        end
+
+        it "uses ipa set via ipa_path" do
+          expect(FastlaneCore::UI).to receive(:success).with("Build is verified, have a 🍪.")
+          Fastlane::FastFile.new.parse("lane :test do
+                verify_build(
+                    ipa_path: '#{correctly_signed_ipa}'
+                )
+            end").runner.execute(:test)
+        end
+      end
+
+      describe "Missing ipa file" do
+        it "raises an error if ipa file path set via lane context does not exist" do
+          Fastlane::Actions.lane_context[Fastlane::Actions::SharedValues::IPA_OUTPUT_PATH] = no_such_file
+
+          expect(FastlaneCore::UI).to receive(:user_error!).with("Unable to find ipa file '#{no_such_file}'.").and_call_original
+          expect do
+            Fastlane::FastFile.new.parse("lane :test do
+                verify_build(
+                    ipa_path: '#{no_such_file}'
```

Comment:
Since this test is testing `lane_context` then I think we should just call `verify_build` without passing `ipa_path` here, is that right? If we pass `ipa_path` explicitly that will override the `lane_context` value.

Reply:
Yep, copy-paste error :) thanks for picking it up :)

Reply:
oh, does it loose context when I squash commits and force push?
I am pretty sure this thread of comments was on line 41 before...